libparsec/crates/client/tests/unit/workspace/remove_entry.rs

```diff
@@ -288,3 +293,171 @@ async fn remove_folder_with_local_changes(
+        p_assert_eq!(e.realm_id, wksp1_id);
+        p_assert_eq!(e.entry_id, child_id);
+    });
+    let child_stat = ops.stat_entry_by_id(child_id).await.unwrap();
+    match (entry_type, child_stat) {
+        (
+            "file",
+            EntryStat::File {
+                need_sync,
+                confinement_point,
+                ..
+            },
+        ) => {
+            p_assert_eq!(need_sync, true);
+            p_assert_eq!(confinement_point, None)
+        }
+        (
+            "folder",
+            EntryStat::Folder {
+                need_sync,
+                confinement_point,
+                ..
+            },
+        ) => {
+            p_assert_eq!(need_sync, true);
+            p_assert_eq!(confinement_point, None)
+        }
+        _ => panic!("Expected a {}", entry_type),
+    }
+
+    // Check that parent is not marked as needing sync
+    let parent_stat = ops.stat_entry_by_id(parent_id).await.unwrap();
+    match parent_stat {
+        EntryStat::Folder { need_sync, .. } => {
+            p_assert_eq!(need_sync, false);
+        }
+        _ => panic!("Expected a folder"),
+    }
+
+    // Check that the child is in the parent's children
+    let children = ops.stat_folder_children_by_id(parent_id).await.unwrap();
+    p_assert_eq!(children.len(), 1);
+    let (name, stat) = children[0].clone();
+    p_assert_eq!(name, "test.tmp".parse::<EntryName>().unwrap());
+    match (entry_type, stat) {
+        (
+            "file",
+            EntryStat::File {
+                id,
+                need_sync,
+                confinement_point,
+                ..
+            },
+        ) => {
+            p_assert_eq!(id, child_id);
+            p_assert_eq!(need_sync, true);
+            // TODO: confinement_point should be Some(parent_id)
+            // See: https://github.com/Scille/parsec-cloud/issues/8276
+            p_assert_eq!(confinement_point, None);
+        }
+        (
+            "folder",
+            EntryStat::Folder {
+                id,
+                need_sync,
+                confinement_point,
+                ..
+            },
+        ) => {
+            p_assert_eq!(id, child_id);
+            p_assert_eq!(need_sync, true);
+            // TODO: confinement_point should be Some(parent_id)
+            // See: https://github.com/Scille/parsec-cloud/issues/8276
+            p_assert_eq!(confinement_point, None);
+        }
+        _ => panic!("Expected a {}", entry_type),
+    }
+
+    // Remove child
+    ops.remove_entry(target).await.unwrap();
+
+    // Parent must be updated
+    spy.assert_next(|e: &EventWorkspaceOpsOutboundSyncNeeded| {
+        p_assert_eq!(e.realm_id, wksp1_id);
+        p_assert_eq!(e.entry_id, parent_id);
+    });
+
+    // Check that the child is NOT in the parent's children
+    let children = ops.stat_folder_children_by_id(parent_id).await.unwrap();
+    assert!(children.is_empty());
+}
```

Comment:
Also verify that the `LocalFolderManifest` is correctly updated by using `WorkspaceStore` directly.

What need to be added is the check for local_confinment_points is empty (since you've only added a single file that was confined).

Since you have access to the local manifest, you could also replace the stat_folder by directly checking the `children` field in the manifest.